ocs_ci/ocs/acm/acm.py

```diff
@@ -625,3 +629,28 @@ def import_clusters_with_acm():
         )
     else:
         import_clusters_via_cli(clusters)
+
+
+class RunWithConfigContext(object):
+    def __init__(self, config_index):
+        self.original_config_index = config.cur_index
+        self.config_index = config_index
+
+    def __enter__(self):
+        config.switch_ctx(self.config_index)
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        config.switch_ctx(self.original_config_index)
+
+
+class RunWithAcmConfigContext(RunWithConfigContext):
+    def __init__(self):
+        acm_index = get_all_acm_indexes()[0]
+        super().__init__(acm_index)
+
+
+class RunWithPrimaryConfigContext(RunWithConfigContext):
+    def __init__(self):
+        primary_index = get_primary_cluster_config()
+        super().__init__(primary_index)
```

Comment:
I would rather see those in config class as those would be reused for other scenarios like provider/client mode.

https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/framework/__init__.py#L25

So user might access those like:

with config.RunWithAcmConfigContext:  # or RunWithConfigContext directly or define new ones leveraging the generic one.

Reply:
done